src/backends/torch/torchdataset.cc

```diff
@@ -366,13 +371,86 @@ namespace dd
     return d.at(i).sizes().vec();
   }
 
+  void TorchDataset::dataaug_then_push(const cv::Mat &bgr,
+                                       const std::vector<torch::Tensor> &t,
+                                       const cv::Mat &bw_target,
+                                       std::vector<BatchToStack> &data,
+                                       std::vector<BatchToStack> &target)
+  {
+    int samples = 1;
+
+    if (_test && _img_rand_aug_cv._crop_params._crop_size > 0)
+      samples = _img_rand_aug_cv._crop_params._test_crop_samples;
+
+    while (samples > 0)
+      {
+        cv::Mat bgr_sample = bgr.clone();
+        cv::Mat bw_target_sample;
+        std::vector<torch::Tensor> d_sample;
+        std::vector<torch::Tensor> t_sample = t;
+        if (_segmentation)
+          bw_target_sample = bw_target.clone();
+
+        // data augmentation can apply here, with OpenCV
+        if (!_test)
+          {
+            if (_bbox)
+              _img_rand_aug_cv.augment_with_bbox(bgr_sample, t_sample);
+            else if (_segmentation)
+              _img_rand_aug_cv.augment_with_segmap(bgr_sample,
+                                                   bw_target_sample);
+            else
+              _img_rand_aug_cv.augment(bgr_sample);
+          }
+        else
+          {
+            // cropping requires test set 'augmentation'
+            if (_bbox)
+              {
+                _img_rand_aug_cv.augment_test_with_bbox(bgr_sample, t_sample);
+              }
+            else if (_segmentation)
+              _img_rand_aug_cv.augment_test_with_segmap(bgr_sample,
+                                                        bw_target_sample);
+            else
+              _img_rand_aug_cv.augment_test(bgr_sample);
+          }
+
+        torch::Tensor imgt
+            = image_to_tensor(bgr_sample, bgr_sample.rows, bgr_sample.cols);
```

Comment:
It looks like "rows" and "cols" parameters are useless, they can be deduced from the cv::Mat. So we can drop them from image_to_tensor?

Reply:
I dont not think it is always the case that we use first param dims when calling image_to_tensor, see for instance torchdataset.cc:260 and torchdataset.cc:279 (ie add_image_batch flavors that are called from inputconn)

Reply:
done